libs/s25main/mapGenerator/Resources.cpp

```diff
@@ -164,9 +164,7 @@ namespace rttr { namespace mapGenerator {
                 map.objectTypes[pt] = tree.index + rnd.RandomValue(0, 7);
             }
 
-            auto graniteProb = probabilities[pt] / 4;
-
-            if(graniteProb > 0 && rnd.ByChance(graniteProb))
+            if(probabilities[pt] > 0 && rnd.ByChance(settings.stonePiles))
```

Comment:
2 questions here: this code may replace a tree that was placed above. Could you change it so that either-or is placed? To keep the current logic this means moving the granite-if before the tree-if and use an else. Simple enough but reads better to have only a single assign to `map.object*`
But main question: Now that the stone probability is mostly independent from `probabilities`, doesn't this mean that stone gets placed almost randomly (especially in small instances instead of chunks)? 
But I'm also concerned about the `probabilities[pt] > 0` (which could reuse `treeProb` from above): If the tree-slider is set to zero then, if I understand the generation of `probabilities` correctly, you will have all zeroes there. So no stone either. IMO that's a (new) bug, because prior the granite and tree prob wasn't configurable. So I'd say for the new setting you need another loop/prop calculation, which is independent from the tree probability. Or am I missing anything?

Reply:
>  this code may replace a tree that was placed above. Could you change it so that either-or is placed? To keep the current logic this means moving the granite-if before the tree-if and use an else.

Just updated that (only using continue instead of if/else). 

> But main question: Now that the stone probability is mostly independent from probabilities, doesn't this mean that stone gets placed almost randomly (especially in small instances instead of chunks)?

Yes, that is right. But granite piles wouldn't really depend on the same parameters as trees. Ideally that would be humidity of the terrain, which is currently implied through the distance to mountain & water. 

I think it makes sense to separate the logic for stone pile generation. And long term I can come up with a more sensible logic (than poor randomness) where to place which amount of stone piles. 

> But I'm also concerned about the probabilities[pt] > 0 (which could reuse treeProb from above): ...

Yes, that's a bug.